oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java

```diff
@@ -132,14 +132,24 @@ public Validator build() {
     }
 
     public void validateAccessToken(String accessToken) {
+        validateAccessToken(accessToken, false);
+    }
+
+    // `at_hash` in ID_TOKEN is mandatory for response_type='id_token token' for Implicit Flow or response_type='code id_token token' for Hybrid Flow.
+    public void validateAccessToken(String accessToken, boolean atHashRequired) {
         if (!Strings.isNullOrEmpty(accessToken)) {
             String atHash = idToken.getClaims().getClaimAsString("at_hash");
             if (Strings.isNullOrEmpty(atHash)) {
-                LOG.warn("Skip access_token validation because corresponding id_token does not have at_hash claim. access_token: " + accessToken + ", id_token: " + idToken);
-                return;
+                if(atHashRequired) {
+                    LOG.error("`at_hash` is missing in `ID_TOKEN`.");
+                    throw new HttpException(ErrorResponseCode.AT_HASH_NOT_FOUND);
```

Comment:
Please add configuration properties `id_token_validation_at_hash_required: true`, ``id_token_validation_c_hash_required: true` and so on. Default value should be `true`.

Reply:
Done